dm/tests/_utils/run_downstream_cluster

```diff
@@ -34,14 +34,15 @@ EOF
 		--data-dir "$WORK_DIR/pd" &
 	# wait until PD is online...
 	i=0
-	while ! curl "http://$PD_ADDR/pd/api/v1/version"; do
+	while curl "http://$PD_ADDR/pd/api/v1/version"; do
```

Comment:
if curl return exit code is not 0, it will not retry such as:

```
Starting PD...

Release Version: v8.5.0-alpha-23-gc8c75510a

Edition: Community

Git Commit Hash: c8c75510a3583c669a962c41e6117157f3d19dca

Git Branch: master

UTC Build Time:  2024-11-14 10:14:13

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 127.0.0.1 port 2379: Connection refused

script returned exit code 2
```

IMO, seems check curl return without error(exit code is 0) is enough🤔
ref tiup: https://github.com/pingcap/tiup/blob/master/pkg/cluster/api/pdapi.go#L188-L195